node/endpoint-handler.js

```diff
@@ -89,16 +90,18 @@ TChannelEndpointHandler.prototype.handleArg1 = function handleArg1(req, buildRes
             'no such endpoint service=%j endpoint=%j',
             req.serviceName, name));
     } else if (handler.canStream) {
-        handler.call(self, req, buildResponse);
+        handler.handleRequest(req, buildResponse);
     } else if (req.streamed) {
-        self.bufferArg23(req, buildResponse, handler);
+        // And implicitly, !handler.canStream, meaning fragmented but
+        // reaccumulated:
+        self.withArg23(req, buildResponse, handler);
     } else {
         res = buildResponse({streamed: false});
-        handler.call(self, req, res, req.arg2, req.arg3);
+        handler.handleRequest(req, res, req.arg2, req.arg3);
     }
 };
 
-TChannelEndpointHandler.prototype.bufferArg23 = function bufferArg23(req, buildResponse, handler) {
+TChannelEndpointHandler.prototype.withArg23 = function withArg23(req, buildResponse, handler) {
```

Comment:
nearly, you'll need a shim of some sort, since:
```
r.withArg23 :: (callback) -> void
callback :: (err, arg2, arg3) -> void
```
In other words, it relies on the fact that its caller had access already to `r` (whether it was a req or res).

So most of the `argsDone` callback function remains, just with a slightly different call signature.  Basically:
- stick with the rename
- use req.withArg23 rather than parallel

As a minor note, this introduces zalgo wrt the type of req (non-streaming calls immed, streaming calls eventually), however I'm 90% sure that all zalgo concerns for handlers v channel have been solved by `TChannelConnectionBase#runHandler` having been deferred circa `connection_base.js:258`.